Capstone M2: Music App

karlavdelgadof opened the pull request:
## Capstone Module #2: Music App :headphones: 

These are the proposed changes :recycle:  for this project :arrow_down: :

### APIs 
- Implemented music [API](https://developer.prod.napster.com/api/v2.2#images-apis) as a development base for the web app :cloud: :musical_note: . The API allows to:
  - Get a list of artists with a unique item id :bust_in_silhouette: :file_cabinet: 
  - For a given item, get detailed information about it :heavy_check_mark: :musical_note: 
- Used provided [Involvement API](https://www.notion.so/microverse/Involvement-API-869e60b5ad104603aa6db59e08150270) to record the different user interactions (likes and comments) :right_anger_bubble: :heart_decoration: 

### Interfaces
- Built Home Page :heavy_check_mark:
- Built comments pop-up :heavy_check_mark:
- Implemented styling based on the provided wireframe :art: 

[Project Video Presentation](https://drive.google.com/file/d/12CNcQwnGTpxRR0-b8-j1QxFcF7ZQf-cA/view?usp=sharing) :movie_camera:

karlavdelgadof committed:
Add gitignore for node modules

karlavdelgadof committed:
Set up linters for the project

karlavdelgadof committed:
Install all webpack dependencies

wayungi committed:
Add jest

wayungi committed:
Install babel

wayungi committed:
Add babel.config file

karlavdelgadof committed:
Merge pull request #22 from karlavdelgadof/set-up

Capstone setup

karlavdelgadof committed:
Update dist files

karlavdelgadof committed:
Add basic html structure and styles

wayungi committed:
Add footer

karlavdelgadof committed:
Add main font to body

karlavdelgadof committed:
Install/add favicon pluggin

karlavdelgadof committed:
Comment line for styling purposes

karlavdelgadof committed:
Fix linter errors

karlavdelgadof committed:
Change tittle to app name

karlavdelgadof committed:
Remove commented code

karlavdelgadof committed:
Replace favicon

karlavdelgadof committed:
Update dist files

karlavdelgadof committed:
Merge pull request #23 from karlavdelgadof/basic-app-structure

Day 2 Capstone M2: Basic app structure

karlavdelgadof committed:
Add functionality draft

karlavdelgadof committed:
Update dist files

karlavdelgadof committed:
Add module for homepage items display

karlavdelgadof committed:
Replace hard coded data and apply modules

karlavdelgadof committed:
Delete comment

karlavdelgadof committed:
Change favicon png

karlavdelgadof committed:
Get data from Napster API

karlavdelgadof committed:
Update styles for homepage

karlavdelgadof committed:
Update dist files

karlavdelgadof committed:
Delete comment

wayungi committed:
Add show comment-popup

wayungi committed:
Add close comment popup button

wayungi committed:
Add getArtistById method

wayungi committed:
Destructure artist data

wayungi committed:
Add image to comment popup

wayungi committed:
Add comment popup title

wayungi committed:
Add artist info on comment popup

wayungi committed:
Clear artist fact div before updating

wayungi committed:
Add comment form

wayungi committed:
Add postComments method

wayungi committed:
Add comment display method

wayungi committed:
Add comment counter

karlavdelgadof committed:
Change hero text

karlavdelgadof committed:
Add min-height to body

karlavdelgadof committed:
Change requested amount of items displayed

karlavdelgadof committed:
Delete unused button

karlavdelgadof committed:
Update dist files

wayungi committed:
Add count comments methods

karlavdelgadof committed:
Correct eslint errors

karlavdelgadof committed:
Delete debugging console log statement

karlavdelgadof committed:
Correct linter error

karlavdelgadof committed:
Merge pull request #26 from karlavdelgadof/homepage-layout-display

Capstone M2: Homepage layout display items (API)

wayungi committed:
Resolve merge conflict

wayungi committed:
Fix merger conflict

wayungi committed:
Remove merger conflict markers

karlavdelgadof committed:
Add event listener for likes icon

wayungi committed:
Update button class

wayungi committed:
Resolve merge conflict

wayungi committed:
Update button class

wayungi committed:
Fix linter errors

wayungi committed:
Fix linter errors

wayungi committed:
Fix merge conflicts

wayungi committed:
Clear form input fields

wayungi committed:
Add 100vh to modal window

wayungi committed:
Create modal window

wayungi committed:
Display modal dynamically

wayungi committed:
Remove black lines

wayungi committed:
Remove comments

wayungi committed:
Fix merger conflicts

wayungi committed:
Update onSubmit event listener

wayungi committed:
Merge pull request #27 from karlavdelgadof/display-comments-popup

Display comments popup

wayungi committed:
Delete commenuPopup function from DOMContentLoaded event

wayungi committed:
Merge branch 'dev' into add-new-comment-2

wayungi committed:
Rename getComments method

wayungi committed:
Fix headers

wayungi committed:
Fix linter errors

wayungi committed:
Fix merge conflicts

wayungi committed:
Change return type to Json

wayungi committed:
Clear commentPopup div content

wayungi committed:
Add title under image

wayungi committed:
Add close button html-entity

wayungi committed:
Add modal window styles

karlavdelgadof committed:
Change fetch url with new app ID

karlavdelgadof committed:
Create class with likes functionality methods

karlavdelgadof committed:
Create styles for likes section

karlavdelgadof committed:
Display likes in the UI

karlavdelgadof committed:
Fix linter errors

karlavdelgadof committed:
Change items limit

karlavdelgadof committed:
Add conditional for 0 likes

karlavdelgadof committed:
Merge branch 'dev' into app-likes-ft

wayungi committed:
Merge pull request #24 from karlavdelgadof/add-new-comment-2

Add new comment 2

karlavdelgadof committed:
Merge branch 'dev' into app-likes-ft

wayungi committed:
Merge branch 'dev' into display-comments-3
Update display-comments-3 with dev changes

karlavdelgadof committed:
Correct linter errors

karlavdelgadof committed:
Merge pull request #28 from karlavdelgadof/add-like-ft

Capstone M2: Add/Display likes

wayungi committed:
Merge branch 'dev' into display-comments-3
Merge likes counting

wayungi committed:
Fix close-modal error

wayungi committed:
Fix linter errors

wayungi committed:
Merge pull request #29 from karlavdelgadof/display-comments-3

Display comments

wayungi committed:
Fix comment counter

karlavdelgadof committed:
Add homepage artist counter feature

karlavdelgadof committed:
Fix linter errors

karlavdelgadof committed:
Merge pull request #31 from karlavdelgadof/items-counter-ft

Capstone M2: Add homepage artist counter feature

wayungi committed:
Merge Item counter
Merge branch 'dev' into add-comments-counter-4

wayungi committed:
Replace innerHTML instances in code

wayungi committed:
Update gitignore file

wayungi committed:
Fix merge conflict

wayungi committed:
Add tests

karlavdelgadof committed:
Add test for items counter

wayungi committed:
Dsiplay comments to page

wayungi committed:
Fix display comments

wayungi committed:
Merge pull request #30 from karlavdelgadof/add-comments-counter-4

Add comments counter

wayungi committed:
Merge pull request #32 from karlavdelgadof/test-branch-5

count comment method test

karlavdelgadof committed:
Change items limit

karlavdelgadof committed:
Update styles

karlavdelgadof committed:
Update styling variables

karlavdelgadof committed:
Ignore coverage folder

karlavdelgadof committed:
Fix linter errors

karlavdelgadof committed:
Update dist files

karlavdelgadof committed:
Merge branch 'dev' into items-test

karlavdelgadof committed:
Update pop-up styles

karlavdelgadof committed:
Correct styles for pop-up

karlavdelgadof committed:
Fix linter errors

karlavdelgadof committed:
Merge pull request #33 from karlavdelgadof/items-test

Capstone M2: Add tests for items counter

karlavdelgadof committed:
Update README file

karlavdelgadof committed:
Update dist files

karlavdelgadof committed:
Correct live demo path

karlavdelgadof committed:
Fix submit comments bug

karlavdelgadof committed:
Add screenshot of the app to README

karlavdelgadof committed:
Fix linter error

medaminedev66 commented on src/modules/createLike.js:
```diff
@@ -0,0 +1,30 @@
+    return response;
+  }
+
+  static async getLikes() {
+    const result = await fetch('https://us-central1-involvement-api.cloudfunctions.net/capstoneApi/apps/q2EecIDolyiDguQmkFIj/likes/');
+    const data = await result.json();
+
+    return data;
+  }
```
- [x] Great job by fetching the likes API endpoint. However, after clicking the like symbol, the likes count is not updated. Please make sure that each time you click the "Like" symbol, the number of likes is updated and shown on the home page.

wayungi replied:
@medaminedev66  change implemented

medaminedev66 commented on src/modules/commentUI.js:
```diff
@@ -0,0 +1,39 @@
+    return parentElement;
+  };
+
+  static showComments = (
+    ele,
+    { creation_date: date, comment: message, username: name },
+  ) => {
+    const {
+      creation_date: creationDate,
+      comment,
+      username,
+    } = { creation_date: date, comment: message, username: name };
+    const comEl = document.createElement('p');
+    comEl.innerText = `${creationDate} ${username} ${comment} `;
+    // console.log(comEl);
+    ele.appendChild(comEl);
+    return comEl;
+  };
```
- [x] The number of comments is not updated after adding a comment. Please ensure that when the user adds a comment, the number of comments changes immediately without closing the pop-up.

wayungi replied:
Done

medaminedev66 commented on src/modules/commentUI.js:
```diff
@@ -0,0 +1,39 @@
+    } = { creation_date: date, comment: message, username: name };
+    const comEl = document.createElement('p');
+    comEl.innerText = `${creationDate} ${username} ${comment} `;
+    // console.log(comEl);
```
- [x] Please remove the commented code to keep your code clean.

medaminedev66 commented on src/modules/commentUI.js:
```diff
@@ -0,0 +1,39 @@
+  static addFacts = async (parentElement, facts) => {
+    // clear old entries
+    parentElement.innerHTML = '';
+    // [type, shortcut, author]
```
- [x] The same thing for this line, try not to commit the commented codes so that your code is clean.

wayungi replied:
Implemented

medaminedev66 commented on src/style.css:
```diff
@@ -1,7 +1,332 @@
+  background-size: cover;
+  backdrop-filter: blur(10px);
+  font-family: 'Source Sans Pro', sans-serif;
+}
```
- [x] Your page design does not fit the desktop screen size. Please try to work more on your design so it fits all the desktop screen sizes. Try to make it flexible as much as you can. Ensure that the page is not scrolling horizontally.
- Your design:
<img width="487" alt="image" src="https://user-images.githubusercontent.com/77190203/181995526-8db5e41d-689d-44db-ac0a-df85b62a2e97.png">

wayungi replied:
Fixed

medaminedev66 reviewed (changes requested):
Hi @karlavdelgadof  and @wayungi 😃, :wave:, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- The repo looks professional
- GitFow best practices have been followed
- There are no linter errors
- You have a descriptive PR
- Your Readme is descriptive
- Your app works as expected and runs without errors

## Required Changes ♻️
- [ ] The video presentation link takes me to a video that is unrelated to your capstone project. Kindly make sure the correct video URL is added to the description of your pull request.

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

wayungi committed:
Fix Likes counter

wayungi committed:
Fix live update of comment count

wayungi committed:
Remove commented code

wayungi committed:
Fix linter errors

wayungi committed:
Fix screen size for desktop

wayungi committed:
Merge branch 'dev' into hotfix-update-likes

wayungi committed:
Fix linter errors

wayungi committed:
Resolve conflict
Merge branch 'hotfix-update-likes' of github.com:karlavdelgadof/Capstone-M2-JS into hotfix-update-likes

wayungi committed:
Fix liner errors

karlavdelgadof committed:
Merge pull request #35 from karlavdelgadof/hotfix-update-likes

Fix Likes counter

wayungi commented:
All requested changes have been implemented @medaminedev66

wayungi committed:
Add video demo link

karlavdelgadof committed:
Merge pull request #36 from karlavdelgadof/hotfix-update-likes

Add video demo link

smunozmo commented on src/modules/commentUI.js:
```diff
@@ -0,0 +1,45 @@
+    return parentElement;
+  };
+
+  static showComments = (
+    ele,
+    { creation_date: date, comment: message, username: name },
+  ) => {
+    const {
+      creation_date: creationDate,
+      comment,
+      username,
+    } = { creation_date: date, comment: message, username: name };
+    const comEl = document.createElement('p');
+    comEl.innerText = `${creationDate} ${username} ${comment} `;
+    ele.appendChild(comEl);
+    return comEl;
+  };
```
- [ ] Guys your project is looking really great, congrats! 🎉 Let's just fix an issue with the comments:

<img width="580" alt="Screen Shot 2022-07-30 at 7 43 17 PM" src="https://user-images.githubusercontent.com/83717285/182005011-dcd0a688-9117-44e4-b0f5-fd1a979fc65b.png">


The comments are stored and loaded properly, however, it's only showing the first 3 comments, if the artist has more than 3 it does not show the comment that you add:

(Also, the comment counter says NaN when you add the first comment)

<img width="634" alt="Screen Shot 2022-07-30 at 7 43 30 PM" src="https://user-images.githubusercontent.com/83717285/182004984-50df69c4-98a2-4864-9c36-4334dcd97c11.png">

Let's try to display all the comments or at least the last comments added but more than three.

karlavdelgadof replied:
@smunozmo Hey, thanks, I've checked the project but in my local host there's no problem so please could you check again? :pray:

smunozmo reviewed (changes requested):
Hi @karlavdelgadof, 👋 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights 👏 

- Great work using the right GitHub Flow.
- Your README looks professional, well done!
- Commits are descriptive, good work.
- Linters are working properly, great job!


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

<img src="https://media1.giphy.com/media/z5cNqbFbMTDA4/giphy.gif"/>

Cheers and Happy coding! 🤓 

✅  Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

⚠️  **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

karlavdelgadof committed:
Update dist files

karlavdelgadof commented:
@smunozmo Here is the demo of my local host and the app is working as it should :+1: 

[Demo](https://www.loom.com/share/81bbe18b1d1b462f869c310d456d5d48) :movie_camera:

mwanawabangona reviewed (approved):
Hi @team!! ,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations!  
![](https://camo.githubusercontent.com/d348d0cdaa16b202354d12402e048f52cd4d589b4ec0e55f07d62c2f22753376/68747470733a2f2f6d65646961322e67697068792e636f6d2f6d656469612f336f373246634a6d4c7a4964594a646d44652f67697068792e676966)

## Highlights
*   No linter errors :heavy_check_mark:
*   Repo looks professional :briefcase:
*   Solid pull request scroll
*   Good quality commits
*   Good job implementing the requested changes!! :clap: :clap: :clap:

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
